lib/AST/ASTContext.cpp

```diff
@@ -1540,10 +1540,12 @@ ASTContext::getXTUDefinition(const FunctionDecl *FD, CompilerInstance &CI,
             iterateContextDecls(TU, MangledFnName, MangleCtx)) {
     llvm::errs() << "Importing function " << MangledFnName << " from "
                  << ASTFileName << "\n";
+
     // FIXME: Refactor const_cast
     auto *ToDecl = cast<FunctionDecl>(
         Importer.Import(const_cast<FunctionDecl *>(ResultDecl)));
-    assert(ToDecl->hasBody());
+    if(!ToDecl || !ToDecl->hasBody())
```

Comment:
In case it is a macro, we should be able to turn it on or off using CMake variables. like -DXTU_ALLOW_FAILED_IMPORT:ON when configuring the project.